strawberryfields/plot.py

```diff
@@ -120,3 +123,235 @@ def generate_wigner_chart(data, xvec, pvec, contours=True):
     chart["layout"]["scene"]["zaxis"]["title"] = "W(x,p)"
 
     return chart
+
+
+# Plot.ly default barchart JSON
+barchart_default = {
+    "data": [{"y": [], "x": [], "type": "bar", "name": "q[0]"}],
+    "layout": {
+        "width": 835,
+        "height": 500,
+        "margin": {"l": 100, "r": 100, "b": 100, "t": 100, "pad": 4},
+        "paper_bgcolor": "rgba(0,0,0,0)",
+        "plot_bgcolor": "rgba(0,0,0,0)",
+        "autosize": True,
+        "yaxis": {
+            "gridcolor": "#bbb",
+            "type": "linear",
+            "autorange": True,
+            "title": "Probability",
+            "fixedrange": True,
+        },
+        "xaxis": {
+            "gridcolor": textcolor,
+            "type": "category",
+            "autorange": True,
+            "title": "q[0]",
+            "fixedrange": True,
+        },
+        "showlegend": False,
+        "annotations": [
+            {
+                "showarrow": False,
+                "yanchor": "bottom",
+                "xref": "paper",
+                "xanchor": "center",
+                "yref": "paper",
+                "text": "",
+                "y": 1,
+                "x": 0,
+                "font": {"size": 16},
+            }
+        ],
+    },
+    "config": {
+        "modeBarButtonsToRemove": ["zoom2d", "lasso2d", "select2d", "toggleSpikelines"],
+        "displaylogo": False,
+    },
+}
+
+
+def plot_fock(state, modes, cutoff=None, renderer="browser"):
+    """Plot the (marginal) Fock state probabilities with Plot.ly.
+
+    Args:
+        state (.BaseState): the state used for plotting
```

Comment:
Always a bit confused, but isn't the correct way like this instead:
```suggestion
        state (:class:`.BaseState`): the state used for plotting
```

Oh, just noticed the review comment from @josh146.

If `.BaseState` is the correct way of doing this, then we probably want to update the Wigner plotting function docstrings to this as well. :slightly_smiling_face:

Reply:
Actually, just checked, and it seems that both make it clickable!

`.BaseState`:
![image](https://user-images.githubusercontent.com/24476053/102535514-51586c00-4076-11eb-9a69-b4ceff3a694b.png)

`:class:`.BaseState``:
![image](https://user-images.githubusercontent.com/24476053/102535603-6c2ae080-4076-11eb-8b8a-3341da593233.png)

Both are clickable and redirect to the correct page. Accepting the suggestion, @josh146 feel free to share if there's a downside missed and will change it back.

Reply:
Oh the second is clickable?! That never used to be the case!